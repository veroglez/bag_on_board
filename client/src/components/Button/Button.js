import React from 'react';
import { bool, func, string } from 'prop-types';
// import mix from '../../common/mix';
import styles from './Button.module.scss';

const Button = ({ disabled, id, onClick, text, ...inherit }) => (
  <button
    id={id}
    type="submit"
    className={styles.button}
    onClick={onClick}
    disabled={disabled}
  >
    <span>{text}</span>
  </button>
);

Button.propTypes = {
  id: string,
  disabled: bool,
  icon: bool,
  onClick: func,
  text: string,
};

Button.defaultProps = {
  id: undefined,
  disabled: false,
  icon: undefined,
  onClick: undefined,
  text: undefined,
};

export default Button;
