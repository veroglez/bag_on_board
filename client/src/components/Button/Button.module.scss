@import '../../common/style.scss';

.button {
  background-color: $color;
  border-radius: $borderRadius;
  border: 0;
  color: white;
  font-size: 14px;
  padding: 8px 20px;
  font-family: $font,
}