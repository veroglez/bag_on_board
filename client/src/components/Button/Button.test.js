import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const text = 'Lorem ipsum';
const func = () => 'test';

describe('<Button>', () => {
  it('renders', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {text}', () => {
    const tree = renderer.create(<Button text={text} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {disabled}', () => {
    const tree = renderer.create(<Button disabled text={text} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onClick}', () => {
    const tree = renderer.create(<Button onClick={func} text={text} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
