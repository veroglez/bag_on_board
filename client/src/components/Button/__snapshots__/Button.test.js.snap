// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders 1`] = `
<button
  className="button"
  disabled={false}
  type="submit"
>
  <span />
</button>
`;

exports[`<Button> when {disabled} 1`] = `
<button
  className="button"
  disabled={true}
  type="submit"
>
  <span>
    Lorem ipsum
  </span>
</button>
`;

exports[`<Button> when {onClick} 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  <span>
    Lorem ipsum
  </span>
</button>
`;

exports[`<Button> when {text} 1`] = `
<button
  className="button"
  disabled={false}
  type="submit"
>
  <span>
    Lorem ipsum
  </span>
</button>
`;
